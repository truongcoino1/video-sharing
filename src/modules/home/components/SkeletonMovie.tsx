import React, { memo } from "react";
import Skeleton from "react-loading-skeleton";

const SkeletonMovie = (): JSX.Element => {
  return (
    <div className="movie-item-skeleton flex justify-between mb-12">
      <Skeleton className="w-[350px] min-w-[350px] h-[196px] flex-shrink-0 overflow-hidden relative" />
      <div className="h-[196px] ml-24 flex-grow">
        <Skeleton
          count={1}
          className="h-[24px] flex-shrink-0 overflow-hidden relative"
        />
         <Skeleton
          count={1}
          className="h-[24px] max-w-[200px] flex-shrink-0 overflow-hidden relative"
        />
        <div className="flex items-center my-12">
          <Skeleton className="h-[18px] min-w-[250px] flex-shrink-0 overflow-hidden" />
        </div>
        <Skeleton className="h-[18px] max-w-[200px] flex-shrink-0 overflow-hidden relative" />
        <Skeleton
          count={3}
          className="h-[12px] flex-shrink-0 overflow-hidden relative"
        />
      </div>
    </div>
  );
};

export default memo(SkeletonMovie);
