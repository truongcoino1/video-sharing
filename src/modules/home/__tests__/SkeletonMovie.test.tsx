import { render, cleanup } from "@testing-library/react";
import SkeletonMovie from "../components/SkeletonMovie";

const push = jest.fn();
jest.mock("next/navigation", () => ({
  useRouter: () => ({
    push: push,
  }),
  usePathname: () => "/",
}));

jest.mock("../hooks/useMovies", () => {
  return {
    useMovies: jest.fn().mockReturnValue({
      movies: [],
      getMovies: jest.fn(),
      period: {
        page: 0,
        limited: false,
        pageSize: 20,
        refresh: false,
        loadMore: false,
      },
      setMovies: jest.fn(),
    }),
  };
});


describe("SkeletonMovie", () => {
  afterEach(cleanup);

  it("share should render without crash", () => {
    const result = render(<SkeletonMovie />);
    const elm = result.container.querySelector(".movie-item-skeleton");
    expect(elm).toBeInTheDocument();
  });

  it("should render correct SkeletonMovie", () => {
    const element = render(<SkeletonMovie />);
    expect(element).toMatchSnapshot();
  });
});
