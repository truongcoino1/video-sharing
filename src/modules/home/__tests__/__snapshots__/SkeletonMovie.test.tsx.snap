// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonMovie should render correct SkeletonMovie 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="movie-item-skeleton flex justify-between mb-12"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton w-[350px] min-w-[350px] h-[196px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
        </span>
        <div
          class="h-[196px] ml-24 flex-grow"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton h-[24px] flex-shrink-0 overflow-hidden relative"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton h-[24px] max-w-[200px] flex-shrink-0 overflow-hidden relative"
            >
              ‌
            </span>
            <br />
          </span>
          <div
            class="flex items-center my-12"
          >
            <span
              aria-busy="true"
              aria-live="polite"
            >
              <span
                class="react-loading-skeleton h-[18px] min-w-[250px] flex-shrink-0 overflow-hidden"
              >
                ‌
              </span>
              <br />
            </span>
          </div>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton h-[18px] max-w-[200px] flex-shrink-0 overflow-hidden relative"
            >
              ‌
            </span>
            <br />
          </span>
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton h-[12px] flex-shrink-0 overflow-hidden relative"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton h-[12px] flex-shrink-0 overflow-hidden relative"
            >
              ‌
            </span>
            <br />
            <span
              class="react-loading-skeleton h-[12px] flex-shrink-0 overflow-hidden relative"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="movie-item-skeleton flex justify-between mb-12"
    >
      <span
        aria-busy="true"
        aria-live="polite"
      >
        <span
          class="react-loading-skeleton w-[350px] min-w-[350px] h-[196px] flex-shrink-0 overflow-hidden relative"
        >
          ‌
        </span>
        <br />
      </span>
      <div
        class="h-[196px] ml-24 flex-grow"
      >
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton h-[24px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton h-[24px] max-w-[200px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
        </span>
        <div
          class="flex items-center my-12"
        >
          <span
            aria-busy="true"
            aria-live="polite"
          >
            <span
              class="react-loading-skeleton h-[18px] min-w-[250px] flex-shrink-0 overflow-hidden"
            >
              ‌
            </span>
            <br />
          </span>
        </div>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton h-[18px] max-w-[200px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
        </span>
        <span
          aria-busy="true"
          aria-live="polite"
        >
          <span
            class="react-loading-skeleton h-[12px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton h-[12px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
          <span
            class="react-loading-skeleton h-[12px] flex-shrink-0 overflow-hidden relative"
          >
            ‌
          </span>
          <br />
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
